rpc-client/src/mock_sender.rs

```diff
@@ -99,6 +99,8 @@ impl RpcSender for MockSender {
         request: RpcRequest,
         params: serde_json::Value,
     ) -> Result<serde_json::Value> {
+        // a small sleep to mock latency for io between rpc server and client
+        tokio::time::sleep(Duration::from_millis(1)).await;
```

Comment:
this is breaking the doctests which i guess don't have an async executor. it's the first introduction of `await` in the mock client, so presumably that. i think we'll need a wrapped sync version of `MockSender`, or yolo a `std::thread::sleep()` in and 🙈